src/Plugin/Filter/EntityEmbedFilter.php

```diff
@@ -102,13 +104,24 @@ public function process($text, $langcode) {
             }
 
             $access = $entity->access('view', NULL, TRUE);
-            $access_metadata = CacheableMetadata::createFromObject($access);
-            $entity_metadata = CacheableMetadata::createFromObject($entity);
```

Comment:
LOL, because I couldn't see the very next line:
$result->addCacheableDependency($access);